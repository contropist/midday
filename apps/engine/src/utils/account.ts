export enum AccountType {
  DEPOSITORY = "depository",
  CREDIT = "credit",
  OTHER_ASSET = "other_asset",
  LOAN = "loan",
  OTHER_LIABILITY = "other_liability",
}

export function getType(type: string): AccountType {
  switch (type) {
    case "depository":
      return AccountType.DEPOSITORY;
    case "credit":
      return AccountType.CREDIT;
    default:
      return AccountType.OTHER_ASSET;
  }
}
