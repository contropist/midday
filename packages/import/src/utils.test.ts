import { expect, test } from "bun:test";
import { csvTransformed } from ".";
import { parseCsv } from "./utils";

const csvComma = `Bokförd,Valutadatum,Text,Typ,Insättningar/uttag,Bokfört saldo
2024-03-08,2024-03-08,ELLEVIO AB (PUBL),Betalning (bg/pg),-168.00,1942493.52
2024-03-08,2024-03-08,ELLEVIO AB (PUBL),Betalning (bg/pg),-165.00,1942661.52
2024-03-06,2024-03-06,BANKTJÄNSTER,Annan,-136.00,1942826.52
2024-03-04,2024-03-04,KOSTNAD,Annan,-31.85,1942962.52
2024-03-04,2024-03-04,H04510236587,Internationell betalning,230400.00,1942994.37
2024-03-04,2024-03-03,AWS AMAZON C/24-03-02,Kortköp,-2797.17,1712594.37
2024-03-04,2024-03-02,GOOGLE  YOUT/24-03-01,Kortköp,-119.00,1715391.54
2024-03-04,2024-03-02,DUBLIN      /24-03-01,Kortköp,-119.40,1715510.54
2024-03-04,2024-03-02,DUBLIN      /24-03-01,Kortköp,-179.11,1715629.94
2024-03-04,2024-03-02,GSUITE WDLIN/24-03-01,Kortköp,-29.00,1715809.05
2024-03-04,2024-03-02,GOOGLE CLOUD/24-03-01,Kortköp,-39.85,1715838.05
2024-03-01,2024-03-01,AMSTERDAM   /24-03-01,Kortköp,-153.10,1715877.90`;

const csvColon = `"Kontonummer";"Kontonamn";"";"Saldo";"Tillgängligt belopp"
"90248745117";"Rörelsekonto";"";"195 460,31";"195 460,31"

"Bokföringsdatum";"Transaktionsdatum";"Transaktionstyp";"Meddelande";"Belopp"
"2024-03-08";"2024-03-08";"Kortköp";"APL*APPLE SEARCH ADS,800-275-2273,US";"-387,27"
"2024-03-08";"2024-03-08";"Betalning";"SKATTEVERKET";"-44 134,00"
"2024-03-08";"2024-03-08";"Överföring";"TILL SKV";"44 134,00"
"2024-03-08";"2024-03-08";"Betalning";"hallon / Billogram";"-485,00"
"2024-03-07";"2024-03-06";"Kortköp";"WEBHALLENE";"-8 007,00"
"2024-03-07";"2024-03-07";"Utlandsbetalning";"ANK Utland Normal";"2 590,96"
"2024-03-06";"2024-03-06";"Överföring";"Skatt moms pwm";"-48 000,00"
"2024-03-06";"2024-03-06";"BG-insättning";"52762176";"137 500,00"
"2024-03-06";"2024-03-06";"Utlandsbetalning";"ANK Utland Normal";"1 923,95"
"2024-03-05";"2024-03-05";"Kortköp";"FIGMA MONTHLY RENEWAL,SAN FRANCISCO,US";"-316,17"
"2024-03-05";"2024-03-04";"Kortköp";"PAYPAL *APPERS GMBH,35314369001,AT";"-806,18"
"2024-03-02";"2024-03-01";"Kortköp";"GOOGLE*CLOUD 5NHT7Z,CC GOOGLE.COM,IE";"-1,22"
"2024-03-02";"2024-03-01";"Kortköp";"GSUITE,Dublin,IE";"-100,00"`;

const csvColo2 = `Datum;Kategori;Underkategori;Text;Belopp;Saldo;Status;Avst�mt
2024-03-10;;;Okq8;-93,00;;V�ntar;Nej
2024-03-10;;;Non Solo Bar P;-220,50;;V�ntar;Nej
2024-03-10;;;Max Burgers 2010140_ki;-134,00;;V�ntar;Nej
2024-03-10;;;Cherry 2 21400;-400,00;;V�ntar;Nej
2024-03-10;;;Cherry 2 21400;-400,00;;V�ntar;Nej
2024-03-10;;;Cherry 2 21400;-400,00;;V�ntar;Nej
2024-03-10;;;Ballbreaker;-676,00;;V�ntar;Nej
2024-03-10;;;Okq8;-74,00;;V�ntar;Nej
2024-03-10;;;Circle K Linkoping Mal;-81,00;;V�ntar;Nej
2024-03-10;;;La Kantarell;-145,00;;V�ntar;Nej
2024-03-10;�vriga inkomster;Konto�verf�ringar;Swish fr�n Gustav Karl Valdemar Alsteryd;1�200,00;12�605,18;Utf�rd;Nej
2024-03-10;�vriga utgifter;�vriga konto�verf�ringar;Swish till Christoffer Collin;-1�325,00;11�405,18;Utf�rd;Nej
2024-03-09;�vriga utgifter;�vriga konto�verf�ringar;Swish till CARL OHLSSON;-110,00;12�730,18;Utf�rd;Nej
2024-03-09;�vriga utgifter;�vriga konto�verf�ringar;Swish till EASYPARK AB;-78,78;12�840,18;Utf�rd;Nej
2024-03-08;Mat & dryck;Livsmedelsaff�r;Shifudo Sverige Ab   ))));-144,00;12�918,96;Utf�rd;Nej
2024-03-08;N�je & fritid;Kaf� & restaurang;Q                    ))));-90,00;13�062,96;Utf�rd;Nej
2024-03-08;Boende;�vrigt;Mimosa Blommor       ))));-345,00;13�152,96;Utf�rd;Nej
2024-03-07;N�je & fritid;Kaf� & restaurang;Shish Kebab          ))));-100,00;13�497,96;Utf�rd;Nej
2024-03-07;Mat & dryck;Livsmedelsaff�r;Hemkop Linkoping Vas ))));-92,95;13�597,96;Utf�rd;Nej
2024-03-07;N�je & fritid;Bio, konsert & teater;Filmstaden Link      ))));-109,00;13�690,91;Utf�rd;Nej
2024-03-07;�vriga utgifter;Kabel-TV;Disney Plus;-119,00;13�799,91;Utf�rd;Nej
2024-03-07;�vriga utgifter;Fackf�rening;Unionen;-235,00;13�918,91;Utf�rd;Nej
2024-03-06;Mat & dryck;Livsmedelsaff�r;Hemkop Linkoping Vas ))));-273,60;14�153,91;Utf�rd;Nej`;

const csvDot = `Bokförd.Valutadatum.Text.Typ.Insättningar/uttag.Bokfört saldo
2024-05-17.2024-05-17.SAN FRANCISC/24-05-16.Kortköp.-714,26.70001,16
2024-05-15.2024-05-15.SKATTEVERKET.Betalning (bg/pg).-67749,00.70715,42
2024-05-08.2024-05-08.VOLKSWAGEN FINANS SVERIG.Betalning (bg/pg).-6615,00.138464,42
2024-05-07.2024-05-07."PREMIE FÖRS.".Annan.-5000,00.145079,42
2024-05-06.2024-05-06.TWITTER PAID/24-05-05.Kortköp.-166,00.150079,42
2024-05-06.2024-05-06.BANKTJÄNSTER.Annan.-264,25.150245,42
2024-05-06.2024-05-06.KOSTNAD.Annan.-31,85.150509,67
2024-05-06.2024-05-06.H06513143882.Internationell betalning.150000,00.150541,52`;

test("Parse from comma delimiter", () => {
  expect(parseCsv(csvComma)).toMatchSnapshot();
});

test("Parse from colon delimiter", () => {
  expect(parseCsv(csvColon)).toMatchSnapshot();
});

test("Transform transactions", () => {
  expect(
    csvTransformed({
      teamId: "123",
      extracted: [
        {
          date: "2024-03-03",
          description: "AWS AMAZON C/24-03-02",
          amount: -2797.17,
        },
        {
          date: "2024-03-02",
          description: "GOOGLE YOUT/24-03-01",
          amount: -119,
        },
        {
          date: "2024-03-04",
          description: "DUBLIN /24-03-01",
          amount: -119.4,
        },
        {
          date: "2024-03-02",
          description: "DUBLIN /24-03-01",
          amount: -179.11,
        },
        {
          date: "2024-03-02",
          description: "GSUITE WDLIN/24-03-01",
          amount: -29,
        },
        {
          date: "2024-03-02",
          description: "GOOGLE CLOUD/24-03-01",
          amount: -39.85,
        },
        {
          date: "2024-03-01",
          description: "AMSTERDAM /24-03-01",
          amount: -153.1,
        },
        {
          date: "2024-02-29",
          description: "TELE2 SV AB FÖRETAG ATTN",
          amount: -971,
        },
        {
          date: "2024-02-27",
          description: "SAN FRANCISC/24-02-26",
          amount: -42.47,
        },
        {
          date: "2024-02-27",
          description: "PARIS /24-02-26",
          amount: -1061.65,
        },
      ],
      raw: csvComma,
    })
  ).toMatchSnapshot();
});

test("Transform transactions", () => {
  expect(
    csvTransformed({
      teamId: "123",
      extracted: [
        { date: "2024-03-08", description: "Överföring", amount: 44134 },
        { date: "2024-03-08", description: "Betalning", amount: -485 },
        { date: "2024-03-07", description: "Kortköp", amount: -8007 },
        { date: "2024-03-06", description: "Överföring", amount: -48000 },
        { date: "2024-03-05", description: "Kortköp", amount: -316.17 },
        { date: "2024-03-05", description: "Kortköp", amount: -806.18 },
        { date: "2024-03-02", description: "Kortköp", amount: -1.22 },
        { date: "2024-03-02", description: "Kortköp", amount: -100 },
        {
          date: "2024-03-01",
          description: "APL*APPLE SEARCH ADS,800-275-2273,US",
          amount: -282.88,
        },
        { date: "2024-03-01", description: "Betalning", amount: -25000 },
        {
          date: "2024-03-01",
          description: "Avgift Utlandsbetalning",
          amount: -30,
        },
      ],
      raw: csvColon,
    })
  ).toMatchSnapshot();
});

test("Transform transactions", () => {
  expect(
    csvTransformed({
      teamId: "123",
      extracted: [
        { date: "2024-03-10", description: "Cherry 2 21400", amount: -400 },
        { date: "2024-03-10", description: "Ballbreaker", amount: -676 },
        { date: "2024-03-10", description: "Okq8", amount: -74 },
        {
          date: "2024-03-10",
          description: "Circle K Linkoping Mal",
          amount: -81,
        },
        { date: "2024-03-10", description: "La Kantarell", amount: -145 },
        {
          date: "2024-03-10",
          description: "Swish från Gustav Karl Valdemar Alsteryd",
          amount: 1200,
        },
        {
          date: "2024-03-10",
          description: "Swish till Christoffer Collin",
          amount: -1325,
        },
        {
          date: "2024-03-09",
          description: "Swish till CARL OHLSSON",
          amount: -110,
        },
        {
          date: "2024-03-09",
          description: "Swish till EASYPARK AB",
          amount: -78,
        },
        {
          date: "2024-03-08",
          description: "Shifudo Sverige Ab",
          amount: -144,
        },
        { date: "2024-03-08", description: "Q", amount: -90 },
        { date: "2024-03-08", description: "Mimosa Blommor", amount: -345 },
        { date: "2024-03-07", description: "Shish Kebab", amount: -100 },
        {
          date: "2024-03-07",
          description: "Hemkop Linkoping Vas",
          amount: -92,
        },
        { date: "2024-03-07", description: "Nöje & fritid; Bio", amount: 0 },
      ],
      raw: csvColo2,
    })
  ).toMatchSnapshot();
});

test("Transform transactions", () => {
  expect(
    csvTransformed({
      teamId: "123",
      extracted: [
        { date: "2024-05-06", description: "BANKTJÄNSTER", amount: -264 },
        { date: "2024-05-06", description: "KOSTNAD Annan", amount: -31 },
        {
          date: "2024-05-06",
          description: "Internationell betalning",
          amount: 150000,
        },
        {
          date: "2024-05-03",
          description: "DUBLIN /24-05-02 Kortköp",
          amount: -99,
        },
        {
          date: "2024-04-30",
          description: "PORSCHE /IF SKADEFÖRSÄKR.Betalning (bg/pg)",
          amount: -3750,
        },
        {
          date: "2024-04-30",
          description: "XENT REDOVISNING AB.Betalning (bg/pg)",
          amount: -3538,
        },
        {
          date: "2024-04-30",
          description: "CITADELLSKAJEN VÄSTRA SA.Betalning (bg/pg)",
          amount: -1875,
        },
        { date: "2024-04-25", description: "LÖN 124913330028", amount: 0 },
        { date: "2024-04-25", description: "Överföring", amount: -38000 },
        {
          date: "2024-04-24",
          description: "SAN FRANCISC/24-04-23 Kortköp",
          amount: -238,
        },
      ],
      raw: csvDot,
    })
  ).toMatchSnapshot();
});
