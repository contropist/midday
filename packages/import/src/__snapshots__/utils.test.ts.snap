// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Parse from comma delimiter 1`] = `
[
  [
    "Bokförd",
    "Valutadatum",
    "Text",
    "Typ",
    "Insättningar/uttag",
    "Bokfört saldo",
  ],
  [
    "2024-03-08",
    "2024-03-08",
    "ELLEVIO AB (PUBL)",
    "Betalning (bg/pg)",
    "-168.00",
    "1942493.52",
  ],
  [
    "2024-03-08",
    "2024-03-08",
    "ELLEVIO AB (PUBL)",
    "Betalning (bg/pg)",
    "-165.00",
    "1942661.52",
  ],
  [
    "2024-03-06",
    "2024-03-06",
    "BANKTJÄNSTER",
    "Annan",
    "-136.00",
    "1942826.52",
  ],
  [
    "2024-03-04",
    "2024-03-04",
    "KOSTNAD",
    "Annan",
    "-31.85",
    "1942962.52",
  ],
  [
    "2024-03-04",
    "2024-03-04",
    "H04510236587",
    "Internationell betalning",
    "230400.00",
    "1942994.37",
  ],
  [
    "2024-03-04",
    "2024-03-03",
    "AWS AMAZON C/24-03-02",
    "Kortköp",
    "-2797.17",
    "1712594.37",
  ],
  [
    "2024-03-04",
    "2024-03-02",
    "GOOGLE  YOUT/24-03-01",
    "Kortköp",
    "-119.00",
    "1715391.54",
  ],
  [
    "2024-03-04",
    "2024-03-02",
    "DUBLIN      /24-03-01",
    "Kortköp",
    "-119.40",
    "1715510.54",
  ],
  [
    "2024-03-04",
    "2024-03-02",
    "DUBLIN      /24-03-01",
    "Kortköp",
    "-179.11",
    "1715629.94",
  ],
  [
    "2024-03-04",
    "2024-03-02",
    "GSUITE WDLIN/24-03-01",
    "Kortköp",
    "-29.00",
    "1715809.05",
  ],
  [
    "2024-03-04",
    "2024-03-02",
    "GOOGLE CLOUD/24-03-01",
    "Kortköp",
    "-39.85",
    "1715838.05",
  ],
  [
    "2024-03-01",
    "2024-03-01",
    "AMSTERDAM   /24-03-01",
    "Kortköp",
    "-153.10",
    "1715877.90",
  ],
]
`;

exports[`Parse from colon delimiter 1`] = `
[
  [
    "Kontonummer",
    "Kontonamn",
    "",
    "Saldo",
    "Tillgängligt belopp",
  ],
  [
    "90248745117",
    "Rörelsekonto",
    "",
    "195 460,31",
    "195 460,31",
  ],
  [
    "Bokföringsdatum",
    "Transaktionsdatum",
    "Transaktionstyp",
    "Meddelande",
    "Belopp",
  ],
  [
    "2024-03-08",
    "2024-03-08",
    "Kortköp",
    "APL*APPLE SEARCH ADS,800-275-2273,US",
    "-387,27",
  ],
  [
    "2024-03-08",
    "2024-03-08",
    "Betalning",
    "SKATTEVERKET",
    "-44 134,00",
  ],
  [
    "2024-03-08",
    "2024-03-08",
    "Överföring",
    "TILL SKV",
    "44 134,00",
  ],
  [
    "2024-03-08",
    "2024-03-08",
    "Betalning",
    "hallon / Billogram",
    "-485,00",
  ],
  [
    "2024-03-07",
    "2024-03-06",
    "Kortköp",
    "WEBHALLENE",
    "-8 007,00",
  ],
  [
    "2024-03-07",
    "2024-03-07",
    "Utlandsbetalning",
    "ANK Utland Normal",
    "2 590,96",
  ],
  [
    "2024-03-06",
    "2024-03-06",
    "Överföring",
    "Skatt moms pwm",
    "-48 000,00",
  ],
  [
    "2024-03-06",
    "2024-03-06",
    "BG-insättning",
    "52762176",
    "137 500,00",
  ],
  [
    "2024-03-06",
    "2024-03-06",
    "Utlandsbetalning",
    "ANK Utland Normal",
    "1 923,95",
  ],
  [
    "2024-03-05",
    "2024-03-05",
    "Kortköp",
    "FIGMA MONTHLY RENEWAL,SAN FRANCISCO,US",
    "-316,17",
  ],
  [
    "2024-03-05",
    "2024-03-04",
    "Kortköp",
    "PAYPAL *APPERS GMBH,35314369001,AT",
    "-806,18",
  ],
  [
    "2024-03-02",
    "2024-03-01",
    "Kortköp",
    "GOOGLE*CLOUD 5NHT7Z,CC GOOGLE.COM,IE",
    "-1,22",
  ],
  [
    "2024-03-02",
    "2024-03-01",
    "Kortköp",
    "GSUITE,Dublin,IE",
    "-100,00",
  ],
]
`;

exports[`Transform transactions 1`] = `
[
  {
    "amount": -168,
    "category": null,
    "date": "2024-03-08",
    "internal_id": "123_360407f43551cbcf0da545ec7e429ce2c69fb1f27c03bb336ff3e4d9c092ec79",
    "manual": true,
    "method": "other",
    "name": "Ellevio Ab Publ",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -165,
    "category": null,
    "date": "2024-03-08",
    "internal_id": "123_360407f43551cbcf0da545ec7e429ce2c69fb1f27c03bb336ff3e4d9c092ec79",
    "manual": true,
    "method": "other",
    "name": "Ellevio Ab Publ",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -136,
    "category": null,
    "date": "2024-03-06",
    "internal_id": "123_9646dac117237a79e9479e2b22935b5d9ff0e63e8be24ce9d84160a12ec4910f",
    "manual": true,
    "method": "other",
    "name": "Banktjänster",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -31.85,
    "category": null,
    "date": "2024-03-04",
    "internal_id": "123_b6c737875c2778e9dabdeb8ca41f67f01443e640856ab2bcc5d16c63bfcc6485",
    "manual": true,
    "method": "other",
    "name": "Kostnad",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": 230400,
    "category": "income",
    "date": "2024-03-04",
    "internal_id": "123_b6c737875c2778e9dabdeb8ca41f67f01443e640856ab2bcc5d16c63bfcc6485",
    "manual": true,
    "method": "other",
    "name": "H04510236587",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -2797.17,
    "category": null,
    "date": "2024-03-03",
    "internal_id": "123_cd621e57e48f3665ddceb17f797d3304bfc443dbd3db862082454986c5015931",
    "manual": true,
    "method": "other",
    "name": "Aws Amazon C 24 03 02",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -119,
    "category": null,
    "date": "2024-03-02",
    "internal_id": "123_8b769aa1612687d676cf0c0a1f819d6dbb230e682d057be174baeed01ae6b68f",
    "manual": true,
    "method": "other",
    "name": "Google Yout 24 03 01",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -119.4,
    "category": null,
    "date": "2024-03-02",
    "internal_id": "123_8b769aa1612687d676cf0c0a1f819d6dbb230e682d057be174baeed01ae6b68f",
    "manual": true,
    "method": "other",
    "name": "Dublin 24 03 01",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -179.11,
    "category": null,
    "date": "2024-03-02",
    "internal_id": "123_8b769aa1612687d676cf0c0a1f819d6dbb230e682d057be174baeed01ae6b68f",
    "manual": true,
    "method": "other",
    "name": "Dublin 24 03 01",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -29,
    "category": null,
    "date": "2024-03-02",
    "internal_id": "123_8b769aa1612687d676cf0c0a1f819d6dbb230e682d057be174baeed01ae6b68f",
    "manual": true,
    "method": "other",
    "name": "Gsuite Wdlin 24 03 01",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -39.85,
    "category": null,
    "date": "2024-03-02",
    "internal_id": "123_8b769aa1612687d676cf0c0a1f819d6dbb230e682d057be174baeed01ae6b68f",
    "manual": true,
    "method": "other",
    "name": "Google Cloud 24 03 01",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -153.1,
    "category": null,
    "date": "2024-03-01",
    "internal_id": "123_846bfdc78358be3ab9f89e0f180d04e86d77842b05c00d9fadcd3bdd408f3cae",
    "manual": true,
    "method": "other",
    "name": "Amsterdam 24 03 01",
    "status": "posted",
    "team_id": "123",
  },
]
`;

exports[`Transform transactions 2`] = `
[
  {
    "amount": -387.27,
    "category": null,
    "date": "2024-03-08",
    "internal_id": "123_360407f43551cbcf0da545ec7e429ce2c69fb1f27c03bb336ff3e4d9c092ec79",
    "manual": true,
    "method": "other",
    "name": "Kortköp",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -44134,
    "category": null,
    "date": "2024-03-08",
    "internal_id": "123_360407f43551cbcf0da545ec7e429ce2c69fb1f27c03bb336ff3e4d9c092ec79",
    "manual": true,
    "method": "other",
    "name": "Betalning",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": 44134,
    "category": "income",
    "date": "2024-03-08",
    "internal_id": "123_360407f43551cbcf0da545ec7e429ce2c69fb1f27c03bb336ff3e4d9c092ec79",
    "manual": true,
    "method": "other",
    "name": "Överföring",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -485,
    "category": null,
    "date": "2024-03-08",
    "internal_id": "123_360407f43551cbcf0da545ec7e429ce2c69fb1f27c03bb336ff3e4d9c092ec79",
    "manual": true,
    "method": "other",
    "name": "Betalning",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -8007,
    "category": null,
    "date": "2024-03-07",
    "internal_id": "123_b03f934a4d02c9e87ac338790993906525f34d1620c55353b2f1901fb1839c8a",
    "manual": true,
    "method": "other",
    "name": "Kortköp",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": 2590.96,
    "category": "income",
    "date": "2024-03-07",
    "internal_id": "123_b03f934a4d02c9e87ac338790993906525f34d1620c55353b2f1901fb1839c8a",
    "manual": true,
    "method": "other",
    "name": "Utlandsbetalning",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -48000,
    "category": null,
    "date": "2024-03-06",
    "internal_id": "123_9646dac117237a79e9479e2b22935b5d9ff0e63e8be24ce9d84160a12ec4910f",
    "manual": true,
    "method": "other",
    "name": "Överföring",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": 137500,
    "category": "income",
    "date": "2024-03-06",
    "internal_id": "123_9646dac117237a79e9479e2b22935b5d9ff0e63e8be24ce9d84160a12ec4910f",
    "manual": true,
    "method": "other",
    "name": "Bg Insättning",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": 1923.95,
    "category": "income",
    "date": "2024-03-06",
    "internal_id": "123_9646dac117237a79e9479e2b22935b5d9ff0e63e8be24ce9d84160a12ec4910f",
    "manual": true,
    "method": "other",
    "name": "Utlandsbetalning",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -316.17,
    "category": null,
    "date": "2024-03-05",
    "internal_id": "123_b56b602b96051d392ea031b9dce383df0404a1e9dda2912488b6ceeb7b8811b8",
    "manual": true,
    "method": "other",
    "name": "Kortköp",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -806.18,
    "category": null,
    "date": "2024-03-05",
    "internal_id": "123_b56b602b96051d392ea031b9dce383df0404a1e9dda2912488b6ceeb7b8811b8",
    "manual": true,
    "method": "other",
    "name": "Kortköp",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -1.22,
    "category": null,
    "date": "2024-03-02",
    "internal_id": "123_8b769aa1612687d676cf0c0a1f819d6dbb230e682d057be174baeed01ae6b68f",
    "manual": true,
    "method": "other",
    "name": "Kortköp",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -100,
    "category": null,
    "date": "2024-03-02",
    "internal_id": "123_8b769aa1612687d676cf0c0a1f819d6dbb230e682d057be174baeed01ae6b68f",
    "manual": true,
    "method": "other",
    "name": "Kortköp",
    "status": "posted",
    "team_id": "123",
  },
]
`;

exports[`Transform transactions 3`] = `
[
  {
    "amount": -93,
    "category": null,
    "date": "2024-03-10",
    "internal_id": "123_dcb80eeaef5ace3667dedb4bff0672ddd6f108c9a4f4127423f41e69ee9d308b",
    "manual": true,
    "method": "other",
    "name": "Okq8",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -220.5,
    "category": null,
    "date": "2024-03-10",
    "internal_id": "123_dcb80eeaef5ace3667dedb4bff0672ddd6f108c9a4f4127423f41e69ee9d308b",
    "manual": true,
    "method": "other",
    "name": "Non Solo Bar P",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -134,
    "category": null,
    "date": "2024-03-10",
    "internal_id": "123_dcb80eeaef5ace3667dedb4bff0672ddd6f108c9a4f4127423f41e69ee9d308b",
    "manual": true,
    "method": "other",
    "name": "Max Burgers 2010140 Ki",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -400,
    "category": null,
    "date": "2024-03-10",
    "internal_id": "123_dcb80eeaef5ace3667dedb4bff0672ddd6f108c9a4f4127423f41e69ee9d308b",
    "manual": true,
    "method": "other",
    "name": "Cherry 2 21400",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -400,
    "category": null,
    "date": "2024-03-10",
    "internal_id": "123_dcb80eeaef5ace3667dedb4bff0672ddd6f108c9a4f4127423f41e69ee9d308b",
    "manual": true,
    "method": "other",
    "name": "Cherry 2 21400",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -400,
    "category": null,
    "date": "2024-03-10",
    "internal_id": "123_dcb80eeaef5ace3667dedb4bff0672ddd6f108c9a4f4127423f41e69ee9d308b",
    "manual": true,
    "method": "other",
    "name": "Cherry 2 21400",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -676,
    "category": null,
    "date": "2024-03-10",
    "internal_id": "123_dcb80eeaef5ace3667dedb4bff0672ddd6f108c9a4f4127423f41e69ee9d308b",
    "manual": true,
    "method": "other",
    "name": "Ballbreaker",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -74,
    "category": null,
    "date": "2024-03-10",
    "internal_id": "123_dcb80eeaef5ace3667dedb4bff0672ddd6f108c9a4f4127423f41e69ee9d308b",
    "manual": true,
    "method": "other",
    "name": "Okq8",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -81,
    "category": null,
    "date": "2024-03-10",
    "internal_id": "123_dcb80eeaef5ace3667dedb4bff0672ddd6f108c9a4f4127423f41e69ee9d308b",
    "manual": true,
    "method": "other",
    "name": "Circle K Linkoping Mal",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -145,
    "category": null,
    "date": "2024-03-10",
    "internal_id": "123_dcb80eeaef5ace3667dedb4bff0672ddd6f108c9a4f4127423f41e69ee9d308b",
    "manual": true,
    "method": "other",
    "name": "La Kantarell",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": 1200,
    "category": "income",
    "date": "2024-03-10",
    "internal_id": "123_dcb80eeaef5ace3667dedb4bff0672ddd6f108c9a4f4127423f41e69ee9d308b",
    "manual": true,
    "method": "other",
    "name": "Swish Fr N Gustav Karl Valdemar Alsteryd",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -1325,
    "category": null,
    "date": "2024-03-10",
    "internal_id": "123_dcb80eeaef5ace3667dedb4bff0672ddd6f108c9a4f4127423f41e69ee9d308b",
    "manual": true,
    "method": "other",
    "name": "Swish Till Christoffer Collin",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -110,
    "category": null,
    "date": "2024-03-09",
    "internal_id": "123_c4132d0d19f20e356cba02253790576cc4009ddebe1a9f7ab8d4969783357acc",
    "manual": true,
    "method": "other",
    "name": "Swish Till Carl Ohlsson",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -78.78,
    "category": null,
    "date": "2024-03-09",
    "internal_id": "123_c4132d0d19f20e356cba02253790576cc4009ddebe1a9f7ab8d4969783357acc",
    "manual": true,
    "method": "other",
    "name": "Swish Till Easypark Ab",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -144,
    "category": null,
    "date": "2024-03-08",
    "internal_id": "123_360407f43551cbcf0da545ec7e429ce2c69fb1f27c03bb336ff3e4d9c092ec79",
    "manual": true,
    "method": "other",
    "name": "Shifudo Sverige Ab",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -90,
    "category": null,
    "date": "2024-03-08",
    "internal_id": "123_360407f43551cbcf0da545ec7e429ce2c69fb1f27c03bb336ff3e4d9c092ec79",
    "manual": true,
    "method": "other",
    "name": "Q",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -345,
    "category": null,
    "date": "2024-03-08",
    "internal_id": "123_360407f43551cbcf0da545ec7e429ce2c69fb1f27c03bb336ff3e4d9c092ec79",
    "manual": true,
    "method": "other",
    "name": "Mimosa Blommor",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -100,
    "category": null,
    "date": "2024-03-07",
    "internal_id": "123_b03f934a4d02c9e87ac338790993906525f34d1620c55353b2f1901fb1839c8a",
    "manual": true,
    "method": "other",
    "name": "Shish Kebab",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -92.95,
    "category": null,
    "date": "2024-03-07",
    "internal_id": "123_b03f934a4d02c9e87ac338790993906525f34d1620c55353b2f1901fb1839c8a",
    "manual": true,
    "method": "other",
    "name": "Hemkop Linkoping Vas",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -109,
    "category": null,
    "date": "2024-03-07",
    "internal_id": "123_b03f934a4d02c9e87ac338790993906525f34d1620c55353b2f1901fb1839c8a",
    "manual": true,
    "method": "other",
    "name": "Filmstaden Link",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -119,
    "category": null,
    "date": "2024-03-07",
    "internal_id": "123_b03f934a4d02c9e87ac338790993906525f34d1620c55353b2f1901fb1839c8a",
    "manual": true,
    "method": "other",
    "name": "Disney Plus",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -235,
    "category": null,
    "date": "2024-03-07",
    "internal_id": "123_b03f934a4d02c9e87ac338790993906525f34d1620c55353b2f1901fb1839c8a",
    "manual": true,
    "method": "other",
    "name": "Unionen",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -273.6,
    "category": null,
    "date": "2024-03-06",
    "internal_id": "123_9646dac117237a79e9479e2b22935b5d9ff0e63e8be24ce9d84160a12ec4910f",
    "manual": true,
    "method": "other",
    "name": "Hemkop Linkoping Vas",
    "status": "posted",
    "team_id": "123",
  },
]
`;

exports[`Transform transactions 4`] = `[]`;
